internal/internal.go

```diff
@@ -15,15 +15,54 @@
 package internal
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"os"
 	"os/user"
 	"path/filepath"
 	"runtime"
 )
 
-// DefaultEndpointFile is the location where the
+// Service contains metadata about the exporter service.
+type Service struct {
+	Endpoint string `json:"endpoint"`
+}
+
+// WriteToEndpointFile writes service metadata to
+// canonical endpoint file.
+func (s *Service) WriteToEndpointFile() (path string, err error) {
+	data, err := json.Marshal(&s)
```

Comment:
Do you need to pass a pointer to a pointer here?

Reply:
Hoho, no. Fixed.